app/src/main/res/layout/activity_main.xml

```diff
@@ -13,11 +13,20 @@
         app:appLogo="@drawable/ic_logo"
         app:icon1="@drawable/ic_notifications"
         app:icon2="@drawable/ic_search"
-        app:layout_constraintBottom_toTopOf="@id/home_page_frame_content"
+        app:layout_constraintBottom_toTopOf="@id/home_page_filter_bar"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
+    <com.noalecohen.dispatcher.customview.CustomFilterBar
+        android:id="@+id/home_page_filter_bar"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintBottom_toTopOf="@id/home_page_frame_content"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/home_page_header" />
+
     <androidx.fragment.app.FragmentContainerView
```

Comment:
instead of creating the extra screen_layout.
You could have changed this element, constraints so it would take the full screen when the user in the search flow.
and revert it when he exits the search flow,